### PR TITLE
Replace environment variable usage with explicit config

### PR DESCRIPTION
**Pull request by zachmargolis**
I discovered the `AuthenticationClient` which is kind of singleton-y and I'm not a fan.... but I aimed for the smallest change I could make instead of trying to refactor it to separate logic from the caching it has

In our current usage, we never use Proofer::Agent, we instantiate clients directly (inside identity-idp-functions) so this should be fairly straightforward

### CODE REVIEW

File: lib/aamva/request/security_token_request.rb [hunk 1]
```diff
@@ -37,8 +38,8 @@ def send
         raise ::Proofer::TimeoutError, message
       end
 
-      def self.auth_url
-        Env.fetch('AAMVA_AUTH_URL', DEFAULT_AUTH_URL)
+      def auth_url
+        config.auth_url || DEFAULT_AUTH_URL
```
Comment (mitchellhenke) [inline lib/aamva/request/security_token_request.rb:42]: can we put the default where the config is loaded?

File: lib/aamva/request/security_token_request.rb [hunk 2]
```diff
@@ -121,7 +122,7 @@ def uuid
       end
 
       def timeout
-        ENV.fetch('AAMVA_AUTH_REQUEST_TIMEOUT', 5).to_i
+        (config.auth_request_timeout || 5).to_i
```
Comment (mitchellhenke) [inline lib/aamva/request/security_token_request.rb:125]: same here, I think we ideally set up the config as much as possible at the beginning?

File: lib/aamva/request/security_token_request.rb (hunk 1 above)
Comment (zachmargolis) [reply lib/aamva/request/security_token_request.rb:42]: Do you mean inline this method?

Comment (mitchellhenke) [reply lib/aamva/request/security_token_request.rb:42]: sorry, misunderstood how it worked. this is good 🙂

**Review by mitchellhenke — APPROVED** [review]
👏🏼 👏🏼 👏🏼 👏🏼 👏🏼

**Review by jmhooper — APPROVED** [review]
:100:

**Comment by jmhooper** [comment]
And yeah, the authentication client is not well put together and I suspect is implicated in many of the timeouts we see. We should remember to come back and revisit that.

**Pull request merged**